bundles/org.eclipse.ui.views.properties.tabbed/src/org/eclipse/ui/views/properties/tabbed/TabbedPropertySheetPage.java

```diff
@@ -325,16 +325,15 @@ protected void handlePartActivated(IWorkbenchPart part) {
 		 * of these events since we want to send aboutToBeHidden() and
 		 * aboutToBeShown() when the property sheet is hidden or shown.
 		 */
-		if (!thisActivated && !part.equals(contributor)
+		if (!thisActivated && !(part instanceof ITabbedPropertySheetPageContributor p && p.equals(contributor))
 				&& !part.getSite().getId().equals(contributor.getContributorId())) {
 			/*
 			 * Is the part is a IContributedContentsView for the contributor,
 			 * for example, outline view.
 			 */
 			IContributedContentsView view = Adapters.adapt(part, IContributedContentsView.class);
-			if (view == null
```

Comment:
It looks like removing the `view ==null` case is causing a lot of NPEs in certain situations:
- https://github.com/eclipse-platform/eclipse.platform.ui/pull/2275

Reply:
And I don't know if the case `view.getContributingPart() == null` is relevant, but before if that was true the if's condition was false, now it is true.

Reply:
it was != null condition.
cp instanceof Something=> cp!=null

Reply:
> cp instanceof Something=> cp!=null

That's right, but the problem is the negation is now applied to the result of the `&&` and not to its RHS.
Instead of 
`!(view.getContributingPart() instanceof ITabbedPropertySheetPageContributor cp && cp.equals(contributor))`
it should probably be
`view.getContributingPart() instanceof ITabbedPropertySheetPageContributor cp && !cp.equals(contributor)`